### PR TITLE
make_file_id: use pageId for generating file ID if possible

### PR DESCRIPTION
**Pull request by kba**
Implements #825

### CODE REVIEW

File: ocrd_utils/ocrd_utils/str.py [hunk 1]
```diff
@@ -61,16 +61,19 @@ def make_file_id(ocrd_file, output_file_grp):
...
+    is not unique, append a counter and increment it until it is unique.
+    Otherwise: Use the position of the ``ocrd_file`` in its fileGrp as a counter and increment it until the
+    concatenation of ``output_file_grp`` with the counter is unique
     """
     ret = ocrd_file.ID.replace(ocrd_file.fileGrp, output_file_grp)
     if ret == ocrd_file.ID:
-        if ocrd_file.pageId and ocrd_file.pageId in ocrd_file.ID:
-            # still sufficiently unique
-            ret = output_file_grp + '_' + ocrd_file.ID
+        if ocrd_file.pageId:
```
Comment (bertsky) [inline ocrd_utils/ocrd_utils/str.py:71]:
```suggestion
        elif ocrd_file.pageId:
```

**Review by bertsky — CHANGES_REQUESTED** [review]
This does not address the issue, though: If you run on an already existing fileGrp, regardless of `--overwrite` or not, you will get new file IDs (for the same pageId) which will be henceforth be ignored (in the default `skip` strategy of `Processor.input_files`).

See #861 for a different (if unfinished) attempt.

But we probably also have to enforce this _outside_ of `make_file_id`, probably in `Workspace.add_file`: When adding a PAGE-XML file for a pageId that already exists in that fileGrp,
- if `overwrite`, then re-use that fileId
- otherwise raise an exception

**Comment by bertsky** [comment]
So can this be closed @kba?

**Comment by kba** [comment]
> So can this be closed @kba?

Yes, I just merged #861

**Pull request closed without merging**